### PR TITLE
Fixed a concurrent access issue in the logs auditor

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Added a lock to the auditor to ensure the registry is thread safe.

### Motivation

Ensuring the agent does not crash while discovery new containers

### CODE REVIEW

File: pkg/logs/auditor/auditor.go [hunk 1]
```diff
@@ -34,35 +35,35 @@ type Auditor struct {
 	inputChan    chan message.Message
 	registry     map[string]*RegistryEntry
 	registryPath string
-
-	entryTTL time.Duration
-
-	done chan struct{}
+	mu           *sync.Mutex
```
Comment (LeoCavaille) [inline pkg/logs/auditor/auditor.go]: Why are you making that a pointer?

Comment (ajacquemot) [reply pkg/logs/auditor/auditor.go]: You're right, it does not have to be a pointer.

File: pkg/logs/auditor/auditor.go [hunk 2]
```diff
@@ -34,35 +35,35 @@ type Auditor struct {
...
+	entryTTL     time.Duration
+	done         chan struct{}
 }
 
 // New returns an initialized Auditor
 func New(inputChan chan message.Message, runPath string) *Auditor {
 	return &Auditor{
 		inputChan:    inputChan,
 		registryPath: filepath.Join(runPath, "registry.json"),
+		mu:           sync.Mutex{},
```
Comment (LeoCavaille) [inline pkg/logs/auditor/auditor.go]: The zero value for a Mutex is an unlocked Mutex, so you don't need to initialize it.

**Review by tmichelet — DISMISSED** [review]
Looks good, thanks.
For next time, could you do the moves / refacto in a separate commit? It's hard to distinguish what has been moved around than what has been updated

**Review by LeoCavaille — APPROVED** [review]

**Review by tmichelet — APPROVED** [review]

**Pull request merged**